cmd/agent/agent.go

```diff
@@ -460,6 +463,103 @@ func serverWSCommunication(config config.Config, messages chan *streaming.WSMess
 	}
 }
 
+func kubeEvents(kubeEnv *agent.KubeEnv, gimletHost string, agentKey string) {
+	namespaces, err := kubeEnv.Client.CoreV1().Namespaces().List(context.TODO(), meta_v1.ListOptions{})
+	if err != nil {
+		log.Errorf("could not get namespaces: %v", err)
+		return
+	}
+
+	svc, err := kubeEnv.Client.CoreV1().Services("default").List(context.TODO(), meta_v1.ListOptions{})
+	if err != nil {
+		log.Errorf("could not get services: %v", err)
+		return
+	}
+
+	var integratedServices []v1.Service
+	for _, s := range svc.Items {
+		if _, ok := s.ObjectMeta.GetAnnotations()[agent.AnnotationGitRepository]; ok {
+			integratedServices = append(integratedServices, s)
+		}
+	}
+
+	var allEvents []api.Event
+	for _, n := range namespaces.Items {
+		allDeployments, err := kubeEnv.Client.AppsV1().Deployments(n.Name).List(context.TODO(), meta_v1.ListOptions{})
```

Comment:
No need to get the namespaces. The `kubeEnv.Client.AppsV1().Deployments(n.Name)` call with an empty string parameter gets deployments from all namespaces